addon/instance-initializers/prefetch.js

```diff
@@ -39,6 +41,14 @@ export function initialize(instance) {
         fullParams.queryParams = transition.queryParams;
       }
 
+      // Skip handlers that have already run with the same params.
+      if (JSON.stringify(routeParamsMap[handlerInfo.name]) === JSON.stringify(fullParams)) {
```

Comment:
This should use [json-stable-stringify](https://github.com/substack/json-stable-stringify).